packages/preview-middleware-client/src/utils/version.ts

```diff
@@ -46,42 +71,40 @@ export async function getUi5Version(): Promise<Ui5VersionInfo> {
  * Checks if the given version is lower than the required minimal version.
  * @param ui5VersionInfo to check
  * @param minUi5VersionInfo to check against (default is 1.71)
+ * @throws Error if the version info is invalid
  *
  * @returns boolean
  */
 export function isLowerThanMinimalUi5Version(
     ui5VersionInfo: Ui5VersionInfo,
     minUi5VersionInfo: Ui5VersionInfo = minVersionInfo
 ): boolean {
-    if (!isNaN(ui5VersionInfo.major) && !isNaN(ui5VersionInfo.minor)) {
-        if (ui5VersionInfo.major < minUi5VersionInfo.major) {
-            return true;
-        }
-        if (ui5VersionInfo.major === minUi5VersionInfo.major && ui5VersionInfo.minor < minUi5VersionInfo.minor) {
-            return true;
-        }
-    }
-    return false;
+    checkVersionInfo(ui5VersionInfo);
+    checkVersionInfo(minUi5VersionInfo);
+    const ui5VersionString = `${ui5VersionInfo.major}.${ui5VersionInfo.minor}.${ui5VersionInfo.patch ?? 0}`;
+    const minUi5VersionString = `${minUi5VersionInfo.major}.${minUi5VersionInfo.minor}.${minUi5VersionInfo.patch ?? 0}`;
+    return lt(ui5VersionString, minUi5VersionString);
 }
 
 /**
  * Checks if the given version is equal to the specified version.
```

Comment:
We should mention it here that patch versions for a specific library might not match the framework/core version and what steps should be taken to ensure that version check produces the desired result.